File: client-go/examples/basic_usage.go
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	"github.com/your-org/soulseekr"
)

func main() {
	// Initialize client
	client := soulseekr.NewClient("http://localhost:8080", "your-api-key-here")
	client.HTTPClient.Timeout = 30 * time.Second

	ctx := context.Background()

	// Check server health
	health, err := client.Health(ctx)
	if err != nil {
		log.Fatalf("Error checking health: %v", err)
	}
	fmt.Printf("Server status: %v\n", health)

	// Get API version
	version, err := client.Version(ctx)
	if err != nil {
		log.Fatalf("Error getting version: %v", err)
	}
	fmt.Printf("API version: %v\n", version)

	// Get capabilities
	caps, err := client.GetCapabilities(ctx)
	if err != nil {
		log.Fatalf("Error getting capabilities: %v", err)
	}
	if endpoints, ok := caps["endpoints"].([]interface{}); ok {
		fmt.Printf("Endpoints available: %d\n", len(endpoints))
	}

	// Get server stats
	stats, err := client.GetStats(ctx)
	if err != nil {
		log.Fatalf("Error getting stats: %v", err)
	}
	fmt.Printf("Total searches: %v\n", stats["total_searches"])
	fmt.Printf("Active transfers: %v\n", stats["active_transfers"])

	// List searches
	searches, err := client.ListSearches(ctx, 10, 0)
	if err != nil {
		log.Fatalf("Error listing searches: %v", err)
	}
	fmt.Printf("\nFound %d recent searches\n", len(searches))
	for i, s := range searches {
		if i >= 3 {
			break
		}
		fmt.Printf("  - Query: %v\n", s["query"])
		fmt.Printf("    Status: %v\n", s["status"])
	}

	// Create a new search
	search, err := client.CreateSearch(ctx, "beethoven symphony")
	if err != nil {
		log.Fatalf("Error creating search: %v", err)
	}
	fmt.Printf("\nCreated search: %v\n", search["id"])
	fmt.Printf("Query: %v\n", search["query"])

	// List messages
	messages, err := client.ListMessages(ctx, 5, 0)
	if err != nil {
		log.Fatalf("Error listing messages: %v", err)
	}
	fmt.Printf("\nMessages: %d\n", len(messages))
	for i, m := range messages {
		if i >= 2 {
			break
		}
		fmt.Printf("  - From: %v\n", m["username"])
		fmt.Printf("    Content: %v\n", m["content"])
	}

	// List transfers
	transfers, err := client.ListTransfers(ctx, "", "", 5, 0)
	if err != nil {
		log.Fatalf("Error listing transfers: %v", err)
	}
	fmt.Printf("\nTransfers: %d\n", len(transfers))
	for i, t := range transfers {
		if i >= 2 {
			break
		}
		fmt.Printf("  - File: %v\n", t["filename"])
		fmt.Printf("    Status: %v\n", t["status"])
	}

	// List users
	users, err := client.ListUsers(ctx, 5, 0)
	if err != nil {
		log.Fatalf("Error listing users: %v", err)
	}
	fmt.Printf("\nUsers: %d\n", len(users))

	// List rooms
	rooms, err := client.ListRooms(ctx)
	if err != nil {
		log.Fatalf("Error listing rooms: %v", err)
	}
	fmt.Printf("Rooms: %d\n", len(rooms))

	// List shares
	shares, err := client.ListShares(ctx, 5, 0)
	if err != nil {
		log.Fatalf("Error listing shares: %v", err)
	}
	fmt.Printf("Shares: %d\n", len(shares))
}

File: client-python/examples/basic_usage.py
#!/usr/bin/env python3
"""
Basic usage example for slskr Python client
"""

import asyncio
from slskr import SlskrClient


async def main():
    """Basic client usage example"""
    
    # Initialize client
    client = SlskrClient(
        base_url="http://127.0.0.1:5030",
        token="your-api-key-here",
        debug=True  # Enable debug logging
    )
    
    try:
        # Check server health
        health = await client.health()
        print(f"Server status: {health.get('status')}")
        
        # Get API version
        version = await client.version()
        print(f"API version: {version.get('version')}")
        
        # Get capabilities
        caps = await client.get_capabilities()
        print(f"Endpoints available: {len(caps.get('endpoints', []))}")
        
        # Get server stats
        stats = await client.get_stats()
        print(f"Total searches: {stats.get('total_searches')}")
        print(f"Active transfers: {stats.get('active_transfers')}")
        
        # List searches
        searches = await client.list_searches(limit=10)
        print(f"\nFound {len(searches)} recent searches")
        for search in searches[:3]:
            print(f"  - Query: {search.get('query')}")
            print(f"    Status: {search.get('status')}")
        
        # Create a new search
        new_search = await client.create_search(query="beethoven symphony")
        print(f"\nCreated search: {new_search.get('id')}")
        print(f"Query: {new_search.get('query')}")
        
        # Get search details
        search_id = new_search.get('id')
        details = await client.get_search_details(search_id, limit=20)
        print(f"Search results: {len(details.get('results', []))}")
        
        # List messages
        messages = await client.list_messages(limit=5)
        print(f"\nMessages: {len(messages)}")
        for msg in messages[:2]:
            print(f"  - From: {msg.get('username')}")
            print(f"    Content: {msg.get('content')[:50]}...")
        
        # List transfers
        transfers = await client.list_transfers(limit=5)
        print(f"\nTransfers: {len(transfers)}")
        for transfer in transfers[:2]:
            print(f"  - File: {transfer.get('filename')}")
            print(f"    Status: {transfer.get('status')}")
            
    except Exception as e:
        print(f"Error: {e}")
    finally:
        await client.close()


if __name__ == "__main__":
    asyncio.run(main())
